modules/crm_tools/src/Element/RoleIcon.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace Drupal\crm_tools\Element;
+
+use Drupal\Component\Utility\Html as HtmlUtility;
+use Drupal\Core\Render\Element\RenderElement;
+use Drupal\Core\Render\Markup;
+use Drupal\Core\Template\Attribute;
+
+/**
+ * Provides a render element for any HTML tag, with properties and value.
+ *
+ * Properties:
+ * - #icon: The icon name to render.
+ * - #attributes: (array, optional) HTML attributes to apply to the icon. The
+ *   attributes are escaped, see \Drupal\Core\Template\Attribute.
+ * - #color: (string, optional) A string for the background hex color.
+ * - #size: (string, optional) A number string to be used for height and width.
+ *
+ * Usage example:
+ * @code
+ * $build['role'] = [
+ *   '#type' => 'role_icon',
+ *   '#icon' => 'person',
+ * ];
+ * @endcode
+ *
+ * @RenderElement("role_icon")
+ */
+class RoleIcon extends RenderElement {
```

Comment:
Would `OpenIconic`/`open_iconic` be better? Although roles are the only place we currently use it, I could imagine it being used elsewhere?